src/debugAdapter/goDebug.ts

```diff
@@ -276,162 +302,193 @@ class Delve {
+								if (this.onclose) { this.onclose(code); }
+							});
+							this.debugProcess.on('error', (err) => {
+								reject(err);
+							});
+							resolve();
+							return;
 						}
-						this.debugProcess = spawn(getBinPathWithPreferredGopath('go', []), runArgs, { env });
-						this.debugProcess.stderr.on('data', chunk => {
-							let str = chunk.toString();
-							if (this.onstderr) { this.onstderr(str); }
-						});
-						this.debugProcess.stdout.on('data', chunk => {
-							let str = chunk.toString();
-							if (this.onstdout) { this.onstdout(str); }
-						});
-						this.debugProcess.on('close', (code) => {
-							logError('Process exiting with code: ' + code);
-							if (this.onclose) { this.onclose(code); }
-						});
-						this.debugProcess.on('error', (err) => {
-							reject(err);
-						});
-						resolve();
-						return;
 					}
 				}
-			}
-			this.noDebug = false;
-			let serverRunning = false;
+				this.noDebug = false;
+
+				// Get default LoadConfig values according to delve API:
+				// https://github.com/derekparker/delve/blob/c5c41f635244a22d93771def1c31cf1e0e9a2e63/service/rpc1/server.go#L13
+				// https://github.com/derekparker/delve/blob/c5c41f635244a22d93771def1c31cf1e0e9a2e63/service/rpc2/server.go#L423
+				this.loadConfig = launchArgs.dlvLoadConfig || {
+					followPointers: true,
+					maxVariableRecurse: 1,
+					maxStringLen: 64,
+					maxArrayValues: 64,
+					maxStructFields: -1
+				};
 
-			// Get default LoadConfig values according to delve API:
-			// https://github.com/derekparker/delve/blob/c5c41f635244a22d93771def1c31cf1e0e9a2e63/service/rpc1/server.go#L13
-			// https://github.com/derekparker/delve/blob/c5c41f635244a22d93771def1c31cf1e0e9a2e63/service/rpc2/server.go#L423
-			this.loadConfig = launchArgs.dlvLoadConfig || {
-				followPointers: true,
-				maxVariableRecurse: 1,
-				maxStringLen: 64,
-				maxArrayValues: 64,
-				maxStructFields: -1
-			};
+				if (!existsSync(launchArgs.dlvToolPath)) {
+					log(`Couldn't find dlv at the Go tools path, ${process.env['GOPATH']}${env['GOPATH'] ? ', ' + env['GOPATH'] : ''} or ${envPath}`);
+					return reject(`Cannot find Delve debugger. Install from https://github.com/derekparker/delve & ensure it is in your Go tools path, "GOPATH/bin" or "PATH".`);
+				}
 
-			if (mode === 'remote') {
-				this.debugProcess = null;
-				serverRunning = true;  // assume server is running when in remote mode
-				connectClient(port, host);
-				return;
-			}
+				const currentGOWorkspace = getCurrentGoWorkspaceFromGOPATH(env['GOPATH'], dirname);
+				dlvArgs.push(mode || 'debug');
+				if (mode === 'exec') {
+					dlvArgs.push(program);
+				} else if (currentGOWorkspace && env['GO111MODULE'] !== 'on') {
+					dlvArgs.push(dirname.substr(currentGOWorkspace.length + 1));
+				}
+				dlvArgs.push('--headless=true', '--listen=' + launchArgs.host + ':' + launchArgs.port.toString());
+				if (!this.isApiV1) {
```

Comment:
Since we already removed `isApiV1` in the json with `apiVersion`, should we just use that instead?

Reply:
It's legacy but I was hesitant to remove it in case somebody was still relying on it.  @ramya-rao-a do you think we can finally remove this?

Reply:
Its `useApiV1` that has been deprecated
`isApiV1` is a variable maintained only in this file

The support for `useApiV1` has been removed in this PR (look at the first few lines inside the constructor for the `Delve` class. I am fine with this, as there is appropriate messaging in place to tell the user what to do in case of version mismatch when remote debugging